keps/sig-apps/3017-pod-healthy-policy-for-pdb/README.md

```diff
@@ -0,0 +1,429 @@
+	// regardless of whether the criteria in a PDB is met.
+	PodReady PodHealthyPolicy = "PodReady"
+	
+	// PodRunning policy means that pods that are in the Running phase
+	// is considered healthy by the disruption controller, regardless of
+	// whether they are Ready or not. Any pods that are in the Running
+	// phase will be counted when computing "disruptionsAllowed" and
+	// will be subject to the PDB for eviction.
+	PodRunning PodHealthyPolicy = "PodRunning"
+)
+```
+
+### Changes to the disruption controller
+
+The disruption controller will be updated to use `podHealthyPolicy` to
+determine how it should compute the value of `disruptionsAllowed`.
+
+### Changes to the eviction API
+
+The eviction API will be updated to use `podHealthyPolicy` to determine whether
+a pod which is Running but not Ready can be evicted regardless of the value of
+`disruptionsAllowed`. This will only be a behavioral change when users have specified
+a `podHealthyPolicy`, and will not require the actual API to change.
+
+### Test Plan
+
+- Unit and integration tests covering:
+  - The current behavior stays unchanged when the policy is not specified.
+  - Correct behavior for both policies in both the disruption controller and
+    the eviction API.
+  - Feature gate disablement.
+- Verify passing existing E2E and conformance tests for PDBs and Eviction.
+
+### Graduation Criteria
+
+#### Alpha
+
+- Feature gate disabled by default.
+- Unit and integration tests passing.
+
+#### Beta
+
+- Feature gate enabled by default.
+- Existing E2E and conformance tests passing.
+
+#### GA
+
+- Every bug report is fixed.
+- The disruption controller and the eviction API ignores the feature gate.
+
+#### Deprecation
+
+N/A
+
+### Upgrade / Downgrade Strategy
+
+No changes required for existing cluster to use the enhancement.
+
+### Version Skew Strategy
+
+This feature doesn't depend on the version for nodes.
+
+## Production Readiness Review Questionnaire
+
+### Feature Enablement and Rollback
+
+###### How can this feature be enabled / disabled in a live cluster?
+
+- [x] Feature gate (also fill in values in `kep.yaml`)
+    - Feature gate name: PodDisruptionBudgetPodHealthyPolicy
+    - Components depending on the feature gate:
+      - kube-controller-manager
+      - kube-apiserver
+- [ ] Other
+    - Describe the mechanism:
+    - Will enabling / disabling the feature require downtime of the control
+      plane?
+    - Will enabling / disabling the feature require downtime or reprovisioning
+      of a node? (Do not assume `Dynamic Kubelet Config` feature is enabled).
+
+###### Does enabling the feature change any default behavior?
+
+No, the behavior is only changed when users specify the `podHealthyPolicy` in
+the PodDisruptionBudget spec.
+
+###### Can the feature be disabled once it has been enabled (i.e. can we roll back the enablement)?
+
+Yes, in that case the disruption controller and the eviction API will just
+use the default legacy behavior.
```

Comment:
```suggestion
use the default behavior.
```

Reply:
or alternatively current, b/c I doubt we'll change the defaults any time soon.

Reply:
Updated.